### PR TITLE
faster rcnn benchmark [WIP]

### PR DESCRIPTION
**Pull request by Jerryzcn**
Make changes to object detection benchmark, so that different hyper-parameters are used for yolo v3 and faster rcnn.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-439-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-439/4/index.html

File: autogluon/task/object_detection/detector.py [hunk 1]
```diff
@@ -202,12 +192,11 @@ def load(cls, checkpoint):
         model_params = state_dict['model_params']
         classes = state_dict['classes']
 
-        model = get_network(args.meta_arch, args.net, classes, mx.cpu(0))
+        model = get_network(args.meta_arch, args.net, classes, ctx=mx.cpu(0), syncbn=args.syncbn)
```
Comment (zhreshold) [inline autogluon/task/object_detection/detector.py:195]: if cpu is used, syncbn should never become true?

File: autogluon/task/object_detection/pipeline.py [hunk 2]
```diff
@@ -260,6 +260,7 @@ def train(net, train_data, val_data, eval_metric, ctx, args, reporter, final_fit
             else:
                 raise NotImplementedError('%s not implemented.' % args.meta_arch)
             trainer.step(batch_size)
+            mx.nd.waitall()
```
Comment (zhreshold) [inline autogluon/task/object_detection/pipeline.py:263]: why is it required?

File: examples/object_detection/benchmark.py [hunk 3]
```diff
@@ -77,29 +81,37 @@ def get_dataset(args):
...
+        kwargs = {'num_trials': 30, 'epochs': epochs,
+                  'net': ag.Categorical('darknet53', 'mobilenet1.0'), 'meta_arch': args.meta_arch,
+                  'lr': ag.Categorical(1e-2, 5e-3, 1e-3, 5e-4, 1e-4, 5e-5), 'transfer': transfer,
+                  'data_shape': ag.Categorical(320, 416), 'nthreads_per_trial': 16,
+                  'ngpus_per_trial': 8, 'batch_size': 64,
+                  'lr_decay_epoch': ag.Categorical('80,90', '85,95'),
+                  'warmup_epochs': ag.Int(1, 10), 'warmup_iters': ag.Int(250, 1000),
+                  'wd': ag.Categorical(1e-4, 5e-4, 2.5e-4), 'syncbn': ag.Bool(),
+                  'label_smooth': ag.Bool(), 'time_limits': time_limits, 'dist_ip_addrs': []}
+    else:
```
Comment (zhreshold) [inline examples/object_detection/benchmark.py:98]:
```python
elif 'faster_rcnn':
else:
  raise NotImplemented
```

File: autogluon/task/object_detection/detector.py (hunk 1 above)
Comment (Jerryzcn) [reply autogluon/task/object_detection/detector.py:195]: i believe syncbn works with cpu as well

File: autogluon/task/object_detection/pipeline.py (hunk 2 above)
Comment (Jerryzcn) [reply autogluon/task/object_detection/pipeline.py:263]: it is not required, but could save memory

**Comment by szha** [comment]
Job PR-439-5 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-439/5/index.html

**Pull request merged**